pages/understanding-json-schema/reference/non_json_data.md

```diff
@@ -7,84 +7,128 @@ section: docs
 
 <Star label="New in draft 7" />
 
-JSON schema has a set of [keywords](../../learn/glossary#keyword) to describe and optionally validate
-non-JSON data stored inside JSON strings. Since it would be difficult to
-write validators for many media types, JSON schema validators are not
-required to validate the contents of JSON strings based on these
-keywords. However, these keywords are still useful for an application
-that consumes validated JSON.
+JSON schema has a set of [keywords](../../learn/glossary#keyword) to describe and optionally validate non-JSON data stored inside JSON strings that are stored or transmitted. Due to the difficulty in writing validators for all media types, JSON schema validators are not required to validate the contents of JSON strings based on these keywords. However, these keywords are useful for applications that consume validated JSON during the processes of encoding and decoding data.  
```

Comment:
I found the first sentence slightly difficult to comprehend, so I made few changes -  

JSON schema has a set of [keywords](../../learn/glossary#keyword) to describe and optionally validate non-JSON data stored inside JSON strings. Due to the difficulty in writing validators for all media types, JSON schema validators are not required to validate the contents of JSON strings based on these keywords. However, these keywords are useful for applications that use validated JSON to encode and decode data during storage and transmittion.

Note: The suggestion is just a preference not a recommendation based on Style guide. Please feel free to ignore.

Reply:
Thank you for the suggestion, I modified the text to make it more understandable.